test/combinations.jl

```diff
@@ -1,429 +1,365 @@
-# This section tests for:
-# - All supported combinations of integral(f, ::Geometry, ::IntegrationAlgorithm) produce accurate results
-# - Invalid applications of integral aliases (e.g. lineintegral) produce a descriptive error
+"""
+This file includes tests for:
+- Supported combinations {::Geometry, ::IntegrationRule} for `integral(f, geometry, rule)`.
+- Integrand functions returning a `Unitful.Quantity` or a `Vector{Unitful.Quantity}`.
+- Callable objects in place of an integrand function.
+- Unsupported combinations produce a descriptive and useful error message.
+- The appropriate alias function, e.g. `lineintegral` for 1D geometries, works.
+- Invalid applications of integral aliases produce a descriptive and useful error message.
+- (Planned) Usage of non-default DifferentiationMethods.
+"""
+
+#===============================================================================
+                        Test Generation Infrastructure
+===============================================================================#
+
+@testsnippet Combinations begin
+    using LinearAlgebra: norm
+    using Meshes
+    using MeshIntegrals
+    using Unitful
+
+    # Used for testing callable objects as integrand functions
+    struct Callable{F <: Function}
+        f::F
+    end
+    (c::Callable)(p::Meshes.Point) = c.f(p)
+
+    # Stores a testable combination
+    struct TestableGeometry{F <: Function, G <: Geometry, U <: Unitful.Quantity}
+        integrand::F
+        geometry::G
+        solution::U
+    end
+
+    # Used to indicate which features are supported for a particular geometry
+    struct SupportStatus
+        # Alias Functions
+        lineintegral::Bool
+        surfaceintegral::Bool
+        volumeintegral::Bool
+        # IntegrationRules
+        gausskronrod::Bool
+        gausslegendre::Bool
+        hadaptivecubature::Bool
+        # DifferentiationMethods
+        # autoenzyme::Bool
+    end
+
+    # Shortcut constructor for geometries with typical support structure
+    function SupportStatus(sym::Symbol)
+        if sym == :line
+            aliases = Bool.((1, 0, 0))
+            rules = Bool.((1, 1, 1))
+            return SupportStatus(aliases..., rules...)
+        elseif sym == :surface
+            aliases = Bool.((0, 1, 0))
+            rules = Bool.((1, 1, 1))
+            return SupportStatus(aliases..., rules...)
+        elseif sym == :volume
+            aliases = Bool.((0, 0, 1))
+            rules = Bool.((0, 1, 1))
+            return SupportStatus(aliases..., rules...)
+        else
+            error("Unrecognized SupportStatus shortcut $(string(sym))")
+        end
+    end
+
+    function runtests(
+            testable::TestableGeometry,
+            supports::SupportStatus;
+            rtol = sqrt(eps())
+    )
+        # Test alias functions
+        for alias in (lineintegral, surfaceintegral, volumeintegral)
+            # if supports.alias
+            if getfield(supports, first(methods(alias)).name)
```

Comment:
```suggestion
            if getfield(supports, nameof(alias))
```
I think this would be the more natural way to obtain the name of the function as a symbol.